### ISSUE TITLE
NUMA-aware support

### ISSUE THREAD

**Issue by GermanAizek** [minetest/minetest#14307]
### Problem

Performance boost is not optimized for 2 or more CPU on motherboards.

### Solutions

Examples my pull requests:
https://github.com/x64dbg/x64dbg/pull/3272
https://github.com/x64dbg/x64dbg/pull/3307

example detection all threads with NUMA https://github.com/GermanAizek/llvm-project/commit/d1fa25f37631b8b33a71fbe9eb4ea89e3a47b723

### Alternatives

Do you think this project should optimize its work for such boards? Now boards with two processors are already too cheap. NUMA performance is seriously better than single bus memory in single cpu configuration, but code needs to be minor rewrite and optimized.

**Comment by GermanAizek** [comment]
An interesting fact:
Xbox 360 is a hybrid UMA design and the PS3 is a hybrid NUMA.

**Comment by HybridDog** [comment]
Is this Microsoft Windows-specific?

**Comment by GermanAizek** [comment]
> Is this relevant for Microsoft Windows only?

Yep incorrect count threads only on Windows, but NUMA-awareness memory allocation optimization can be done quickly if WinAPI is used. In Linux, there will have to be a lot rewriting.
